tests/src/main/java/fixtures/bodyboolean/Bools.java

```diff
@@ -90,7 +92,7 @@ public Bools(AutoRestBoolTestService client) {
         if (this.client.getHost() == null) {
             throw new IllegalArgumentException("Parameter this.client.getHost() is required and cannot be null.");
         }
-        return service.getTrue(this.client.getHost());
+        return FluxUtil.withContext(context -> service.getTrue(this.client.getHost(), context));
```

Comment:
The way we currently generate in v3 is by adding the context on the method signature: https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/BlobsImpl.java#L273

Reply:
Do you prefer we do this on v4 as well (add Context into client method)?

Reply:
@srnagar may know the correct design for this

Reply:
@srnagar 

Here is sample generated code for simple REST.

Here is sample generated code for paging (withContext is called inside SinglePageAsync and NextSinglePageAsync method).
https://github.com/Azure/autorest.java/blob/3639f7f3d5723e1533de2897f3a58c83a3936d72/tests/src/main/java/fixtures/paging/Pagings.java#L344-L386

Let me know if this is OK for you.

Reply:
I see this pattern is used across the existing SDKs. Not sure why they put it in the FluxUtil but fine.